Documentation/user-guides/alerting.md

```diff
@@ -30,13 +57,87 @@ spec:
   replicas: 3
 ```
 
-The Alertmanager instances will not be able to start up, unless a valid configuration is given. An config file Secret will be composed by taking an optional base config file Secret specified through the `configSecret` field in the Alertmanager resource Spec, and merging that with any AlertmanagerConfig resources that get matched by using the `alertmanagerConfigSelector` and `alertmanagerConfigNamespaceSelector` selectors from the `Alertmanager` resource.
+Wait for all Alertmanager pods to be ready:
+
+```bash
+oc get pods -l alertmanager=main -w
+```
+
+## Managing Alertmanager configuration
+
+By default, the Alertmanager instances will start with a minimal configuration
+which isn't really useful since it doesn't send any notification when receiving
+alerts.
+
+You have several options to provide the [Alertmanager configuration](https://prometheus.io/docs/alerting/configuration/):
+1. You can use a native Alertmanager configuration file stored in a Kubernetes secret.
+2. You can use `spec.alertmanagerConfiguration` to reference an
+   AlertmanagerConfig object in the same namespace which defines the main
+   Alertmanager configuration.
+3. You can define `spec.alertmanagerConfigSelector` and
+   `spec.alertmanagerConfigNamespaceSelector` to tell the operator which
+   AlertmanagerConfigs objects should be selected and merged with the main
+   Alertmanager configuration.
 
-For more information on configuring Alertmanager, see the Prometheus [Alerting Configuration document](https://prometheus.io/docs/alerting/configuration/).
+### Using a Kubernetes Secret
 
-## AlertmanagerConfig Resource
+The following native Alertmanager configuration sends notifications to a fictuous webhook service:
+
+```yaml mdox-exec="cat example/user-guides/alerting/alertmanager.yaml"
+route:
+  group_by: ['job']
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 12h
+  receiver: 'webhook'
+receivers:
+- name: 'webhook'
+  webhook_configs:
+  - url: 'http://example.com/'
+```
+
+Save the above configuration in a file called `alertmanager.yaml` in the local directory and create a Secret from it:
+
+```bash
+kubectl create secret generic alertmanager-example --from-file=alertmanager.yaml
+```
 
-The following example configuration creates an AlertmanagerConfig resource that sends notifications to a non-existent `wechat` receiver:
+The Prometheus operator requires the Secret to be named like
+`alertmanager-{ALERTMANAGER_NAME}`. In the previous example, the name of the
+Alertmanager is `example`, so the secret name must be `alertmanager-example`.
+The name of the key holding the configuration data in the Secret has to be
+`alertmanager.yaml`.
+
+> Note: if you want to use a different secret name, you can specify it with the `spec.configSecret` field in the Alertmanager resource.
```

Comment:
To be honest, I'd almost make this the default explanation and change the current example to a note but its just an opinion and I am content with leaving it also. I'd just lean towards a change because its a similar pattern that might end up being used elsewhere.